zkvm/src/vm.rs

```diff
@@ -693,12 +691,21 @@ where
     }
 
     fn add_range_proof(&mut self, bitrange: usize, expr: Expression) -> Result<(), VMError> {
-        spacesuit::range_proof(
-            self.delegate.cs(),
-            r1cs::LinearCombination::from_iter(expr.terms),
-            ScalarWitness::option_to_integer(expr.assignment)?,
-            bitrange,
-        )
-        .map_err(|_| VMError::R1CSInconsistency)
+        match expr {
```

Comment:
maybe factor out linear combination and assignment?

```rust
let (lc, assignment) = match expr {
    Expression::Constant(x) => (r1cs::LinearCombination::from(x), Some(x)),
    Expression::Terms(terms, assignment) => (r1cs::LinearCombination::from_iter(terms), assignment),
};
spacesuit::range_proof(
    self.delegate.cs(),
    lc, 
    ScalarWitness::option_to_integer(assignment)?,
    bitrange
).map_err(|_| VMError::R1CSInconsistency)
```

Reply:
This is really nice +1